CMakeLists.txt

```diff
@@ -12,7 +12,12 @@ include(GNUInstallDirs)
 
 # The C++ standard whose features are required to build Binaryen.
 # Keep in sync with scripts/test/shared.py cxx_standard
-set(CXX_STANDARD 17)
+# The if condition allows embedding in a project with a higher default C++ standard set
+if((NOT CMAKE_CXX_STANDARD) OR (CMAKE_CXX_STANDARD LESS 17))
```

Comment:
If somebody asked for less that 17 maybe it would be nicer to error out rather than slightly ignore?